### PR TITLE
rocAL -  CMake and header files Clean up

### PR DESCRIPTION
**Pull request by shobana-mcw**
1. Fix issue with Image aug app CMake
2. CMake clean up in rocAL and rocAL_Pybind
3. Header include clean ups.

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
@shobana-mcw This needs a sync up with develop branch

**Review by rrawther — COMMENTED** [review]
It looks like it is not synced and only files changed shows up

**Comment by shobana-mcw** [comment]
Updated with TOT.

File: apps/image_augmentation/CMakeLists.txt [hunk 1]
```diff
@@ -46,7 +46,7 @@ find_package(OpenCV QUIET)
 find_package(AMDRPP QUIET)
 
 # Application Includes & Libraries
-include_directories(${ROCM_PATH}/${CMAKE_INSTALL_INCLUDEDIR}/mivisionx)
+include_directories(${ROCM_PATH}/include/mivisionx/rocal)
```
Comment (LakshmiKumar23) [inline apps/image_augmentation/CMakeLists.txt:49]: @shobana-mcw This cannot be hardcoded. rocm team needs it for spack and other packages

**Review by LakshmiKumar23 — CHANGES_REQUESTED** [review]

File: apps/image_augmentation/CMakeLists.txt (hunk 1 above)
Comment (LakshmiKumar23) [reply apps/image_augmentation/CMakeLists.txt]: @shobana-mcw Yes Shobana, you can do it in this PR if it's just those one line changes. Please test the changes

**Review by LakshmiKumar23 — APPROVED** [review]
@kiritigowda ran the test cases. Ok to merge

**Review by rrawther — APPROVED** [review]
@LakshmiKumar23 or @paveltc  to test it.

**Comment by kiritigowda** [comment]
@LakshmiKumar23 conflict needs to be resolved.

**Comment by LakshmiKumar23** [comment]
> @LakshmiKumar23 conflict needs to be resolved.

@kiritigowda I thought it was resolved. @shobana-mcw Can you resolve the merge conflicts?

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**